common/protos/proto/user.proto

```diff
@@ -96,6 +96,7 @@ message UserOption {
   uint64 min_compatible = 101;
 
   uint64 flags = 1;
+  optional string default_role = 2;
```

Comment:
Increase the `VER` to `VER=3`:
https://github.com/datafuselabs/databend/blob/19a1c73b46b52e2ad628aebf42edacf3345a2c05/common/proto-conv/src/util.rs#L17

Add a test that builds a protobuf message with `VER=3`(this version) and check if it can be deserialized correctly. This test will guarantee the future databend-query works correctly with this `VER=3` data. This test should be kept until `MIN_COMPATIBLE_VER` becomes greater than 3.
https://github.com/datafuselabs/databend/blob/19a1c73b46b52e2ad628aebf42edacf3345a2c05/common/proto-conv/tests/it/user_proto_conv.rs#L181

Reply:
https://github.com/datafuselabs/databend/pull/6847/files#diff-05bf2ee6ffcd5df0b1398460e2ab2d4034062d2a8333946e21691dd5f410e615R118

is this change correct as your describes? q.q